core/beacon/errors.go

```diff
@@ -43,9 +43,10 @@ var (
 
 	INVALIDBLOCKHASH = "INVALID_BLOCK_HASH"
 
-	GenericServerError = rpc.CustomError{Code: -32000, ValidationError: "Server error"}
-	UnknownPayload     = rpc.CustomError{Code: -32001, ValidationError: "Unknown payload"}
-	InvalidTB          = rpc.CustomError{Code: -32002, ValidationError: "Invalid terminal block"}
+	GenericServerError       = rpc.CustomError{Code: -32000, ValidationError: "Server error"}
+	UnknownPayload           = rpc.CustomError{Code: -38001, ValidationError: "Unknown payload"}
+	InvalidForkChoiceState   = rpc.CustomError{Code: -38002, ValidationError: "Invalid forkchoice state"}
+	InvalidPayloadAttributes = rpc.CustomError{Code: -38003, ValidationError: "Invalid payload attributes"}
```

Comment:
This feels like a bad change FWIW. The *code* is already there to "group" the error into `Invalid payload attributes`, but IMHO the `ValidationError` should be the detailed error that Geth returns currently so the other side has a clue what happens (the dev I mean). Otherwise it's "computer says no" effect.

Reply:
Can't we keep the code but use Geth's own errors as the validation error field? Or can't we add a second error reason field to pass it from Geth?

Reply:
Spec says we can add a data field https://github.com/ethereum/execution-apis/blob/main/src/engine/specification.md#errors, we should do that and keep Geth's errors throughout.

Reply:
```
  "error": {
    "code": -32000,
    "message": "Server error",
    "data": {
        "err": "Database corrupted"
    }
  }
```